app/javascript/gdpr/GDPRConsentControls.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+import React, { PureComponent } from 'react';
+import { FormattedMessage } from 'react-intl';
+import classnames from 'classnames';
+import './GDPRConsentControls.css';
+
+type Props = {
+  onChange: (consented: boolean) => void,
+  consented: ?boolean,
+  shortLabels?: boolean,
+};
+
+export class GDPRConsentControls extends PureComponent {
+  props: Props;
+  render() {
+    const { shortLabels, consented } = this.props;
+    const acceptClass = classnames({
+      active: consented === true,
+    });
+    const declineClass = classnames({
+      active: consented === false,
+    });
+    return (
+      <div className="GDPRConsentControls">
+        <label className={acceptClass}>
+          <input
+            type="radio"
+            name="consented"
+            value={1}
+            checked={consented === true}
+            onChange={() => this.props.onChange(true)}
+          />
+          {shortLabels ? (
```

Comment:
Removing `shortLabels` would couple the component with the idea of variants, which are optional and will always be a moving target. It would also mean that we'd need to update the component every time we change or add new variants. Having a list of variants in the component would also make the render method quite noisy the for anything that's more than just a couple of components.

The css class with the variant name is mainly useful for styling during experiments. The actual `input type=radio` is always rendered, but styled differently. It makes it easier to create new templates with new variants that will style them differently, without having to deploy a new version of the component.

Reply:
👍